src/app/dock_widgets/process/gt_processdock.cpp

```diff
@@ -1399,6 +1401,91 @@ GtProcessDock::renameElement()
     }
 }
 
+void
+GtProcessDock::moveElements(const QList<QModelIndex>& source,
+                            const QModelIndex& target)
+{
+    if (!target.isValid() || source.isEmpty()) return;
+
+    QList<QModelIndex> mapped;
+
+    for (auto i : source)
+    {
+        if (i.isValid()) mapped.append(mapToSource(i));
+    }
+
+    if (mapped.isEmpty()) return;
+
+    QList<GtObject*> objToMove;
+
+    /// check if all elements to move have the same parent
+    GtObject* parent = nullptr;
+
+    for (auto j : mapped)
+    {
+        if (j.model() != gtDataModel) return;
+
+        if (auto p = gtDataModel->objectFromIndex(j))
+        {
+            if (!parent) parent = p->parentObject();
+
+            if (parent != p->parentObject())
+            {
+                gtWarning() << tr("It is only allowed to move elements of the "
+                                  "same task");
+                return;
+            }
+
+            objToMove.append(p);
+        }
+    }
+
+    if (objToMove.isEmpty()) return;
+
+    QModelIndex mappedTarget = mapToSource(target);
+
+    GtObject* targetObject = gtDataModel->objectFromIndex(mappedTarget);
+
+    auto targetComp = qobject_cast<GtProcessComponent*>(targetObject);
+
+    if (!targetComp) return;
+
+    auto moveCmd = gtApp->makeCommand(gtApp->currentProject(),
+                                       tr("move process element"));
+
+    if (auto taskParent = qobject_cast<GtTask*>(targetComp))
+    {
+        for (auto o: objToMove)
+        {
+            gtDataModel->appendChild(o, taskParent);
+        }
+    }
+    else
+    {
```

Comment:
Since you dont need to call `moveCmd.finalize()` you can exit early here and do not require an else case
```suggestion
        return;
    }
```